tinygrad/codegen/lowerer.py

```diff
@@ -180,9 +181,24 @@ def _to_uop(self, x:LazyOp) -> UOp:
           vec_load = UOp(UOps.LOAD, load_dtype.vec(4), (buf, idx) + ((UOp.const(load_dtype.vec(4), 0), valid) if has_valid else ()) + barrier)
           return functools.reduce(lambda ret, i: id4.ne(i).where(ret, UOp(UOps.GEP, load_dtype, (vec_load,), i)),
                                   range(4), UOp.const(load_dtype, float('nan')))
+        # NOTE: if there's a following reduceop we need to load the result of this reduceop back into every thread
+        # NOTE: this matches the pattern LOAD -> REDUCE -> STORE -> LOAD
+        load_back = len(x.src) == 1 and x.src[0].op is BufferOps.STORE and \
+          x.src[0].src[0].op in ReduceOps and len(x.src[0].src[0].arg) > 0 and \
+            x.src[0].src[0].src[0].op is BufferOps.LOAD and x.src[0].src[0].src[0].arg.idx < 0
+        if load_back:
+          idx, _ = st_to_uops(x.arg.st, [UOp.const(u.dtype, 0) if i in x.src[0].src[0].arg else u for i,u in enumerate(self.ridxs)], x.arg.dtype)
+          return UOp(UOps.LOAD, load_dtype, (buf, idx) + ((UOp.const(load_dtype, 0), valid) if has_valid else ()) + barrier)
         return UOp(UOps.LOAD, load_dtype, (buf, idx) + ((UOp.const(load_dtype, 0), valid) if has_valid else ()) + barrier)
       # NOTE: only store the local reduceop in the first thread (this is wrong for non group for reduces!)
-      if x.arg.idx >= 0:
+      store_back = \
+        x.arg.idx < 0 and \
+        x.src[0].op in [ReduceOps.SUM, ReduceOps.MAX] and \
```

Comment:
```suggestion
        x.src[0].op in ReduceOps and \
```

Reply:
This is intentional, we don't want `Reduceops.WMMA`.
I can play with it though, this might be redundant with the other parts of the expression

Reply:
but we don't have double WMMA do we? Are you asserting this?